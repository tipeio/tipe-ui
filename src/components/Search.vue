<template>
  <input @click="onClick">
</template>

<script>
export default {
  name: 'TipeSearch',

  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>

<style scoped>
button {
  border: '1px solid #eee';
  border-radius: 3;
  background-color: '#FFFFFF';
  cursor: 'pointer';
  font-size: 15;
  padding: '3px 10px';
  margin: 10;
}
</style>
