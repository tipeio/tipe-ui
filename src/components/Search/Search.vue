<template>
  <div :data-tipe-ui="$options.name">
    <type-icon
      class="search-icon"
      icon="search"
      color="#b5baca"
    />
    <input
      type="search"
      placeholder="Find everthing..."
    >
  </div>
</template>

<script>
import TypeIcon from '../Icon'

export default {
  name: 'TipeSearch',
  components: {
    TypeIcon
  },
  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeSearch'] {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: auto;
}

.search-icon {
  flex: 0 0 auto;
  height: 1.25rem;
  width: 1.25rem;
}

[data-tipe-ui='TipeSearch'] > input {
  flex: 1;
  margin-left: 0.437rem;
  margin-right: 0.437rem;
  border: none;
  border-radius: 0;
  color: var(--text-gray);
  background-color: transparent;
  font-size: 1rem;
  height: 1.625rem;
  line-height: 1.625rem;
}

[data-tipe-ui='TipeSearch'] > input::-webkit-input-placeholder {
  color: var(--text-gray);
  opacity: 0.4;
}
</style>
