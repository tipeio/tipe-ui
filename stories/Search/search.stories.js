import { storiesOf } from '@storybook/vue'
import TipeSearch from '../../src/components/Search'

storiesOf('Search', module).add('default', () => ({
  components: { TipeSearch },
  template: '<tipe-search />'
}))
