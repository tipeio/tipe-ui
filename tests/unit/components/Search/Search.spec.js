import { shallowMount } from '@vue/test-utils'
import Search from '@/components/Search'

describe('Search', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(Search)

      expect(wrapper).toMatchSnapshot()
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(Search)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSearch')
    })
  })
})
