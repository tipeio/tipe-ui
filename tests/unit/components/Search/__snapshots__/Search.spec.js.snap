// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeSearch">
  <typeicon-stub icon="search" height="1rem" width="1rem" color="#000" __iconloader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}" class="search-icon"></typeicon-stub> <input type="search"
    placeholder="Find everthing...">
</div>
`;
